src/ALCProxy.Communication/ALCServer.cs

```diff
@@ -0,0 +1,148 @@
+        {
+            var ci = instanceType.GetConstructor(constructorTypes);
+            instance = (I)ci.Invoke(constructorArgs);
+        }
+        /// <summary>
+        /// Takes a Type that's been passed from the user ALC, and loads it into the current ALC for use. 
+        /// </summary>
+        protected Type ConvertType(Type toConvert)
+        {
+            AssemblyName assemblyName = Assembly.GetAssembly(toConvert).GetName();
+            return currentLoadContext.LoadFromAssemblyName(assemblyName).GetType(toConvert.FullName);
+        }
+        public object CallObject(MethodInfo targetMethod, IList<object> serializedObjects, IList<Type> argTypes)
+        {
+            if (targetMethod == null || serializedObjects.Count != argTypes.Count)
+                throw new ArgumentNullException();
+
+            //Turn the serialized objects into their respective objects
+            argTypes = argTypes.Select(x => ConvertType(x)).ToList();
+            object[] args = DeserializeParameters(serializedObjects, argTypes);
+            MethodInfo[] methods = instanceIntType.GetRuntimeMethods().ToArray();
+            MethodInfo m = FindMethod(methods, targetMethod, argTypes.ToArray());
+            if (m.ContainsGenericParameters)
+            {
+                //While this may work without the conversion, we want it to uphold the type-load boundary, don't let the passed in method use anything from outside the target ALC
+                m = m.MakeGenericMethod(targetMethod.GetGenericArguments());
+            }
+            return SerializeReturnObject(m.Invoke(instance, args), m.ReturnType);
+        }
+        /// <summary>
+        /// Searches for methods within the type to find the one that matches our passed in type. Since the types are technically different,
+        /// using a .Equals() on the methods doesn't have the comparison work correctly, so the first if statement does that manually for us.
+        /// </summary>
+        protected MethodInfo FindMethod(MethodInfo[] methods, MethodInfo targetMethod, Type[] parameterTypes/*These have already been converted so no issues with compatibility*/)
+        {
+            string methodName = targetMethod.Name;
+            foreach (MethodInfo m in methods)
+            {
+                if (!m.Name.Equals(methodName) || parameterTypes.Length != m.GetParameters().Length)
+                {
+                    continue;
+                }
+                bool methodParamsAlligned = true;
+                for (int i = 0; i < parameterTypes.Length; i++)
+                {
+                    if (!RecursivelyCheckForTypes(parameterTypes[i], m.GetParameters()[i].ParameterType))
+                    {
+                        methodParamsAlligned = false;
+                        break;
+                    }
+                }
+                if (!methodParamsAlligned)
+                    continue;
```

Comment:
Add braces whenever in a nested context (even for single line branches). Generally, no braces at the root scope or at the beginning of the method for arg validation is fine.